server/server_test.go

```diff
@@ -49,12 +49,24 @@ var (
 	regression                 = true
 )
 
-func genPort() uint {
-	return uint(atomic.AddUint32(&portGenerator, 1))
+func genPorts() (uint, uint) {
+	var (
+		ports []int
+		err   error
+	)
+	for ports, err = freeport.GetFreePorts(2); err != nil; ports, err = freeport.GetFreePorts(2) {
```

Comment:
Please add the max retry count to avoid possible dead loop.

BTW, why not just follow the example given in the Freeport [homepage](https://github.com/phayes/freeport#golang-example)?

```
port, err := freeport.GetFreePort()
if err != nil {
	log.Fatal(err)
}
```